src/Resources/templates/parameters.php.twig

```diff
@@ -1,3 +1,27 @@
+{#- Formats the default value of a parameter as sane as possible. -#}
+{%- macro defaultValue(parameter) -%}
+    {%- set defaultValue = parameter.defaultValue -%}
+    {%- if defaultValue is same as(true) -%}
+        true
+    {%- elseif defaultValue is same as(false) -%}
+        false
+    {%- elseif parameter.isDefaultValueConstant -%}
+        {%- if not (parameter.getDefaultValueConstantName matches '/^self/') -%}
+            \
+        {%- endif -%}
+        {{- parameter.getDefaultValueConstantName -}}
+    {%- elseif defaultValue is null -%}
+        null
+    {%- elseif defaultValue is iterable -%}
+        []
+    {%- elseif parameter.type.name != "string" and defaultValue matches '/^\\d+(.\\d+){0,1}$/' -%}
+        {{- defaultValue -}}
+    {%- else -%}
+        null {#- Note: this will print $foo = 'bar' as $foo = null -#}
+    {%- endif -%}
+{%- endmacro -%}
```

Comment:
A lot of logic, I would like to unit test in isolation.
Maybe an accessor method on ReflectionParameter?